flang/test/Fir/cse.fir

```diff
@@ -75,3 +75,39 @@ func.func @fun(%arg0: !fir.ref<i64, volatile>) -> i64 {
 // CHECK:           fir.store %[[VAL_3]] to %arg0 : !fir.ref<i64, volatile>
 // CHECK:           return %[[VAL_3]] : i64
 // CHECK:         }
+
+// -----
+
+// Check that volatile hlfir assignments are PRESERVED.
```

Comment:
@jeanPerier Thanks for the review on the other PR! This is the test I added for hlfir assignments. It may be worth adding tests for all/most other hlfir operations as well.